src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,60 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int LOGIN_MIN_LENGTH = 6;
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkIfUserIsNull(user);
+        checkIfUserExists(user);
+        checkIfLoginIsValid(user.getLogin());
+        checkIfPasswordIsValid(user.getPassword());
+        checkIfAgeIsValid(user.getAge());
+        return storageDao.add(user);
```

Comment:
The method `register` is well-structured and follows the checklist by breaking down the validations into separate methods. However, ensure that the `checkIfUserExists` method does not introduce a potential race condition by querying and then adding a user without transactional integrity.